eclipse-collections/src/main/java/org/eclipse/collections/impl/multimap/AbstractSynchronizedMultimap.java

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright (c) 2016 Shotaro Sano.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompany this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.eclipse.collections.impl.multimap;
+
+import java.util.Collection;
+
+import net.jcip.annotations.GuardedBy;
+import org.eclipse.collections.api.RichIterable;
+import org.eclipse.collections.api.bag.Bag;
+import org.eclipse.collections.api.block.function.Function0;
+import org.eclipse.collections.api.block.procedure.Procedure;
+import org.eclipse.collections.api.block.procedure.Procedure2;
+import org.eclipse.collections.api.map.MutableMap;
+import org.eclipse.collections.api.multimap.Multimap;
+import org.eclipse.collections.api.set.SetIterable;
+import org.eclipse.collections.api.tuple.Pair;
+
+public abstract class AbstractSynchronizedMultimap<K, V> implements Multimap<K, V>
+{
+    protected final Object lock;
```

Comment:
There are two protected fields, the lock and the delegate. We access the delegate through methods so that we can cast it to a more specific type. It may be possible to use generics instead, and make the type of `delegate` some `T extends MutableMultimap<K,V>`. I'm not sure if that's possible, or desirable since it increases complexity. But if possible, I think it would be fine to get rid of the getters completely.